src/target/riscv/riscv.h

```diff
@@ -111,6 +111,13 @@ typedef struct {
 	char *name;
 } range_list_t;
 
+struct reg_name;
+
+struct reg_name_table {
+	unsigned int names_num;
+	struct reg_name *reg_names;
+};
+
```

Comment:
For linked list, please consider using the already existing resources from `list.h`.

It would be slightly more code, but there are benefits to it:

- The code would be unified with the rest of OpenOCD.
- Higher chance of being accepted in upstream as is.